boards/posix/native_sim/CMakeLists.txt

```diff
@@ -27,22 +27,9 @@ if(CONFIG_HAS_SDL)
   add_subdirectory(${ZEPHYR_BASE}/boards/${ARCH}/common/sdl/ ${CMAKE_CURRENT_BINARY_DIR}/sdl)
 endif()
 
-set(zephyr_build_path ${CMAKE_BINARY_DIR}/zephyr)
-
 set(nsi_config_content
-  "NSI_BUILD_OPTIONS:=$<JOIN:$<TARGET_PROPERTY:native_simulator,INTERFACE_COMPILE_OPTIONS>,\ >"
-  "NSI_BUILD_PATH:=${zephyr_build_path}/NSI"
-  "NSI_CC:=${CMAKE_C_COMPILER}"
-  "NSI_EMBEDDED_CPU_SW:=${zephyr_build_path}/${KERNEL_ELF_NAME}"
-  "NSI_EXE:=${zephyr_build_path}/${KERNEL_EXE_NAME}"
-  "NSI_EXTRA_SRCS:=$<JOIN:$<TARGET_PROPERTY:native_simulator,INTERFACE_SOURCES>,\ >"
-  "NSI_LINK_OPTIONS:=$<JOIN:$<TARGET_PROPERTY:native_simulator,INTERFACE_LINK_OPTIONS>,\ >"
-  "NSI_PATH:=${NSI_DIR}/"
+  ${nsi_config_content}
   "NSI_NATIVE=1"
 )
 
-string(REPLACE ";" "\n" nsi_config_content "${nsi_config_content}")
-
-file(GENERATE OUTPUT "${zephyr_build_path}/NSI/nsi_config"
-  CONTENT "${nsi_config_content}"
-)
+include(../common/natsim_config.cmake)
```

Comment:
small future proposal.

When doing more work on native simulator, then you may consider to create a function which take arguments instead.

That will also allow more freedom, as calling a function does not rely on the path location of the cmake file.

Comment made, so it's not lost in future, but no changes requested in this PR.